### PR TITLE
Fix commit-details lost on load

### LINKED ISSUE #451: commit-details are not available after loading a ledger from storage
Per an initial issue in fluree/http-api-gateway#25, and per test cases in `history_test` of fluree/db, I'd expect the following message to `/history` to return commit details like `"f:t"` but I'm only getting back `"f:data"` (see below)

```json
{
  "ledger": "example/time-travel",
  "query": {
     "commit-details": true,
     "t": {
        "at": "latest"
      }
   }
}
```
```json
[
    {
        "f:commit": {
            "f:data": {
                "f:assert": [
                    {
                        "ex:name": "Andrew Johnson",
                        "id": "ex:Andrew"
                    }
                ],
                "f:retract": [
                    {
                        "ex:name": "Andrew",
                        "id": "ex:Andrew"
                    }
                ]
            }
        }
    }
]
```

Test case in `history_test`:
https://github.com/fluree/db/blob/main/test/fluree/db/query/history_test.clj#L238-L296
```clojure
        (let [commit-5 {:f/commit
                        {:f/address
                         "fluree:memory://16d376f6cac29e8125ec3beca7aea6f75fb7a4328d73cbe0d629c6132510621c",
                         :f/v       0,
                         :f/previous
                         {:id
                          "fluree:commit:sha256:busykqnrg3i2zdhi7wvlbhretnv32e5cyxcqwvgjtazftkjnpup7"},
                         :f/time    720000,
                         :id
                         "fluree:commit:sha256:bbgkybvkkdwelzlmj6vei7kitm5hvxecotx6imytpta272jwbiaw7",
                         :f/branch  "main",
                         :f/message "meow",
                         :f/data
                         {:f/previous
                          {:id
                           "fluree:db:sha256:bbiioffkgezekkxzojudoqrtk3zmjcxeotecprh2ordafskng7hhc"},
                          :f/address
                          "fluree:memory://783086c375a712eba5d5a74b18882a98a1f77eccf51bddd9b99e2c625e67a088",
                          :f/flakes  102,
                          :f/size    9328,
                          :f/t       5,
                          :f/assert  [{:ex/x "foo-cat", :ex/y "bar-cat", :id :ex/alice}],
                          :f/retract [{:ex/x "foo-3", :ex/y "bar-3", :id :ex/alice}]},
                         "https://www.w3.org/2018/credentials#issuer"
                         {:id "did:fluree:TfCzWTrXqF16hvKGjcYiLxRoYJ1B8a6UMH6"},
                         :f/alias   "committest",
                         :f/context
                         "fluree:memory://b6dcf8968183239ecc7a664025f247de5b7859ac18cdeaace89aafc421eeddee"}}

;; ...

          (is (= [commit-5]
                 @(fluree/history ledger {:commit-details true :t {:at :latest}})))))
```

### PR DESCRIPTION
**Pull request by cap10morgan**
Fixes #451 

We weren't creating the commit details flakes on reification like we were on commits to in-memory ledger values. This refactors that code and then reuses it for reification and fixes some related issues with having access to the necessary values (e.g. commit addresses) to hydrate these flakes on load.

### CODE REVIEW

File: src/fluree/db/json_ld/commit_data.cljc [hunk 1]
```diff
@@ -437,86 +437,144 @@
...
+     (flake/create db-sid const/$_commitdata:t db-t const/$xsd:int t true nil)
+     ;; address
+     (flake/create db-sid const/$_address db-address const/$xsd:string t true nil)
+     ;; size
+     (flake/create db-sid const/$_commitdata:size size const/$xsd:int t true nil)
+     ;; flakes
+     (flake/create db-sid const/$_commitdata:flakes flakes const/$xsd:int t true
+                   nil)]))
+
+(defn prev-commit-flakes
```
Comment (zonotope) [inline src/fluree/db/json_ld/commit_data.cljc:476]: pedantic nitpick that you can take or leave, but it looks like only one flake is created here, in `prev-data-flakes`, and in `message-flakes` below

File: src/fluree/db/json_ld/reify.cljc [hunk 2]
```diff
@@ -302,42 +302,96 @@
...
+         default-ctx-flakes (when defaultContext
+                              (<? (commit-data/default-ctx-flakes
+                                   db t-new next-sid defaultContext)))
+         all-flakes         (-> db
+                                (get-in [:novelty :spot])
+                                empty
+                                (into metadata-flakes)
+                                (into retract-flakes)
+                                (into flakes)
+                                (cond-> prev-commit-flakes
```
Comment (zonotope) [inline src/fluree/db/json_ld/reify.cljc:377]: could we collapse all these `cond->` into one?

**Review by zonotope — APPROVED** [review]
lgtm besides some minor style things you can take or leave.

File: src/fluree/db/json_ld/commit_data.cljc (hunk 1 above)
Comment (cap10morgan) [reply src/fluree/db/json_ld/commit_data.cljc:476]: Yeah it's plural to indicate that it returns a collection and so callers needn't necessarily care how many flakes are returned.

**Pull request merged**